architecture/azure/resources.adoc

```diff
@@ -0,0 +1,132 @@
+[[architecture-azure-resources]]
```

Comment:
@Makdaam I have a number of suggestions and comments on the larger commits. Would you prefer that I write them out or just make updates to this PR?

Reply:
@maxwelldb Feel free to make changes to this PR. If it doesn't change the technical meaning I'm ok with it.